src/dune_rules/colors.ml

```diff
@@ -41,6 +41,13 @@ let mark_open_stag = function
   | _ -> ""
 ;;
 
+let setup_terminal_width () =
+  let env = Env.initial in
+  match Env.mem ~var:"DUNE_TEST__DISABLE_LINE_BREAK" env with
+  | false -> ()
+  | true -> Format.set_margin Format.pp_infinity
```

Comment:
I think you need to set this for stderr as well as stdout

Reply:
I needed to change/replace `User_message.prerr`, indeed, though this code turned out to not do what I hoped for since apparently the global `Format` margin is not respected when printing to strings.